control/controlclient/direct.go

```diff
@@ -143,6 +147,10 @@ type Options struct {
 	// If we receive a new DialPlan from the server, this value will be
 	// updated.
 	DialPlan ControlDialPlanner
+
+	// ControlKeyCache caches Noise keys returned from control; if nil, no
+	// cache will be used.
+	ControlKeyCache cache.Cache[*tailcfg.OverTLSPublicKeyResponse]
```

Comment:
Oh I just realized how this work, I had the wrong view initially

This won't work with FUS when working with profiles across say Tailscale and Headscale. It will cache the first entry and reuse for both instances

Reply:
Hm, it looks like this worked largely due to an implementation detail; `resetForProfileChangeLockedOnEntry` calls `b.Start`, which calls `resetControlClientLockedAsync` and then re-creates the control client with a new in-memory-only cache. This won't work if we switch to a disk cache, though.

Pushed a change that keys based on the actual URL being fetched, instead.